packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -94,6 +94,9 @@ class RunCommand extends RunCommandBase {
     argParser.addOption('use-application-binary',
         hide: !verboseHelp,
         help: 'Specify a pre-built application binary to use when running.');
+    argParser.addOption('kernel',
+        hide: !verboseHelp,
+        help: 'Path to a pre-built kernel blob to use when running.');
```

Comment:
This is hidden because, well, it is not meant for public consumption. It only exists for testing  the new kernel code generation and execution on Flutter and will probably change or be removed once everything switches over from snapshots to kernel blobs. Having just written that, I'll update the code with something along those lines.